fleet_adapter_mir/fleet_adapter_mir/robot_adapter_mir.py

```diff
@@ -0,0 +1,491 @@
+        execution = self.execution
+        if execution is not None and not self.done:
+            return execution.identifier
+        return None
+
+class RobotAdapterMiR:
+    def __init__(
+        self,
+        name: str,
+        rmf_config: rmf_easy.RobotConfiguration,
+        mir_config: dict,
+        # pickup_config: dict,
+        conversions: dict,
+        rmf_missions: dict,
+        fleet_handle,
+        fleet_config,
+        plugin_config: dict | None,
+        node: Node,
+        event_loop,
+        debug=False
+    ):
+        self.name = name
+        self.node = node
+        self.mission: MissionHandle | None = None
+        self.event_loop = event_loop
+        self.debug = debug
+
+        self.disconnect = False
+
+        prefix = mir_config['base_url']
+        headers = {
+            'Content-Type': mir_config['user'],
+            'Authorization': mir_config['password'],
+        }
+        self.api: MirAPI = MirAPI(prefix, headers, conversions, rmf_missions)
+
+        status = self.api.status_get()
+        while self.api.status_get() is None:
+            status = self.api.status_get()
+            time.sleep(0.1)
+        self.last_known_status: MirStatus = status
+        self.current_map = status.state.map
+        self.fleet_config = fleet_config
+
+        self.api.update_known_positions()
+        self.api.positions_delete()
+
+        # This must be done after the API is made
+        self.lift_positions: dict = conversions.get('lift_positions', {})
+        for lift, levels in self.lift_positions.items():
+            for level, position in levels.items():
+                position_name = position.get('name')
+                assert position_name is not None, (
+                    f'Missing "name" field for level [{level}] in lift '
+                    f'[{lift}] for lift_positions in configuration file'
+                )
+                known = self.api.known_positions.get(position_name)
+                assert known is not None, (
+                    f'MiR is missing the position [{position_name}] needed as '
+                    f'a lift position on level [{level}] for lift [{lift}]'
+                )
+        self.nav_issue_ticket = None  # We should only have one issue ticket at a time to manage unsuccessful missions
+        self.requested_replan = False
+        self.replan_counts = 0
+
+        self.fleet_handle = fleet_handle
+        self.update_handle = fleet_handle.add_robot(
+            self.name,
+            status.state,
+            rmf_config,
+            self._make_callbacks(),
+        )
+
+        self.plugins = {}
+        # Available plugins:
+        if 'rmf_cart_delivery' in plugin_config:
+            self.plugins['rmf_cart_delivery'] = CartDelivery(
+                node=node,
+                name=name,
+                mir_api=self.api,
+                update_handle=self.update_handle,
+                actions=plugin_config['rmf_cart_delivery']['actions'],
```

Comment:
Instead of passing this and the next two arguments in as all separate arguments, why not just pass in the `plugin_config` and parse these fields out later?